gsdk/src/storage.rs

```diff
@@ -286,6 +286,26 @@ impl Api {
 
         Ok(mailbox)
     }
+
+    /// Get up to `count` messages from the mailbox
+    pub async fn all_mailbox(&self, count: u32) -> Result<Vec<(StoredMessage, Interval<u32>)>> {
```

Comment:
I'd rename this method to the `mailbox` and current `mailbox` to the `account_mailbox`.

Reply:
Also, we need to filter messages by self account ID as a destination. We are interested in our messages instead of someone else's. We may provide two functions in `gclient` as we do it for one message. E.g., `get_mailbox_messages` (destination is signer) and `get_account_mailbox_messages` (destination is set explicitly as a parameter).